Assets/Scripts/Player/UnitGridCombat.cs

```diff
@@ -260,5 +336,94 @@ private enum State {
 			hasExecuted_ = false;
 			isComplete_ = false;
 		}
+		
+		private void updateAnimation() {
+			if(animator_ == null) {
+				return;
+			}
+
+			if(shouldPlayAttackAnimation_) {

```

Comment:
Not sure if that works with the animations but i think it is easier to make a function were you set all the bools to false and call it first. And after that just set the ones you want to true, the result will be a bit better to read. Basically what the `else` does below.

Reply:
I see what you mean yeah, that should work so I'll do that.